File: 프로그래머스/unrated/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    targets.sort(key=lambda x: (x[0], x[1]))
    chk = 0
    answer = 0
    for s, e in targets:
        if s >= chk:
            answer += 1
            chk = e
        elif e < chk:
            chk = e
    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets) {
    const sortedTargets = targets.sort((a, b) => a[1] - b[1]);
    let answer = 0;
    let destroyedY = -1;
    console.log(sortedTargets);
    for (const [s, e] of sortedTargets) {
        if (destroyedY <= s) {
            destroyedY = e;
            answer += 1;
        };
    };
    return answer;
};